File: climbingStairs.py
class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        if n == 1:
        	return 1
        if n == 2:
        	return 2
        m, l = 1, 2
        for i in range(3, n+1):
        	m, l = l, m + l
        return l

s = Solution()
print s.climbStairs(5)

File: JAVA/climbingStairs.java
public class Solution {
    public int climbStairs(int n) {
        if (n==1)
            return 1;
        int last = 1, cur =2, count=0, temp;
        while (count++<n-2){
            temp = cur;
            cur = cur+last;
            last = temp;
        }
        return cur;
    }
}